hkmc2/shared/src/main/scala/hkmc2/codegen/BlockTransformer.scala

```diff
@@ -147,14 +147,17 @@ class BlockTransformer(subst: SymbolSubst):
     if (own2 is fun.owner) && (sym2 is fun.sym) && (params2 is fun.params) && (body2 is fun.body)
       then fun else FunDefn(own2, sym2, params2, body2)
   
+  def applyValDefn(defn: ValDefn): ValDefn =
```

Comment:
(Only reason I extracetd this is because it used to be needed in my previous refactoring; I kept it for consistency with `FunDefn`.)

Reply:
For consistency, we should also probably extract `applyClsLikeDefn`. Note that this may require some changes where `applyDefn` is overridden.